GetVisibleServices/handler.ts

```diff
@@ -52,11 +54,43 @@ export function GetVisibleServicesHandler(
           `Error getting visible services list: ${error.message}`
         ),
       maybeVisibleServicesJson => {
-        const servicesTuples = onlyNationalService
-          ? toServicesTuple(
-              new StrMap(maybeVisibleServicesJson.getOrElse({}))
-            ).filter(_ => _.scope === ServiceScopeEnum.NATIONAL)
-          : toServicesTuple(new StrMap(maybeVisibleServicesJson.getOrElse({})));
+        const allServicesTuples = toServicesTuple(
+          new StrMap(maybeVisibleServicesJson.getOrElse({}))
+        );
+        const scopedServicesTuples = allServicesTuples.reduce(
+          (acc, service) => {
+            if (service.scope === ServiceScopeEnum.LOCAL) {
+              return {
+                [ServiceScopeEnum.NATIONAL]: acc[ServiceScopeEnum.NATIONAL],
+                [ServiceScopeEnum.LOCAL]: [
+                  ...acc[ServiceScopeEnum.LOCAL],
+                  service
+                ]
+              };
+            }
+            return {
+              [ServiceScopeEnum.NATIONAL]: [
+                ...acc[ServiceScopeEnum.NATIONAL],
+                service
+              ],
+              [ServiceScopeEnum.LOCAL]: acc[ServiceScopeEnum.LOCAL]
+            };
+          },
+          {
+            [ServiceScopeEnum.NATIONAL]: [] as ReadonlyArray<ServiceTuple>,
+            [ServiceScopeEnum.LOCAL]: [] as ReadonlyArray<ServiceTuple>
+          }
+        );
```

Comment:
These instructions are at a lower level of abstraction compared to other steps of the chain, I think we should move them in a separate function, example: `groupByScope(maybeVisibleServicesJson.getOrElse({}))`

Reply:
Refractory with de9dd22. What do you think?

Reply:
I think it's fine, the whole process is much shorter and you don't have to dig into detail to understand what's happening. I left a couple of suggestions, but no one is really important.